bitcoin/src/util/from_hex_str.rs

```diff
@@ -0,0 +1,44 @@
+
+//! FromHexStr
+//! Functionality to allow types to take hex strings as input
+
+#[cfg(feature = "no-std")]
+use crate::prelude::String;
+
+#[cfg(feature = "no-std")]
+use crate::alloc::string::ToString;
+
+
+/// Hex Parsing error
+#[derive(Debug, Eq, PartialEq, Clone)]
+#[non_exhaustive]
+pub enum FromHexError {
+    /// Parsing Error
+    ParseHex(String),
+    /// No prefix was given when calling from_hex_str
+    MissingPrefix(String),
+}
+
+/// Trait that allows types to be initialized from hex strings
+pub trait FromHexStr {
+    /// Parses provided string as hex requiring 0x prefix.
```

Comment:
Missing `type Error;` which I believe we should have in case some integer newtypes have invariants.

Reply:
added